### PR TITLE
95 mentorship index

### LINKED ISSUE #95: Index of all mentorships a person has participated in.
A user should be able to see a list of all the mentorships they have been either a mentee or mentor in.

And on the profile page, it should only display the 3 most recent

Also show the `updated_at` so the user can see whether its a new or old mentorship

### PR DESCRIPTION
**Pull request by mrgenixus**
## Description
Add a list of all previous (and current mentorships)

## Github Issue
Fixes #95

## Screenshots

image

image

## Deploy Notes
*remove if inapplicable*

## Definition of Done

- [x] This PR has appropriate test coverage.

### CODE REVIEW

**Comment by mrgenixus** [comment]
@rebeccafae I seem unable to assign reviewers on this, however it is ready for review, I believe

**Comment by rebeccafae** [comment]
@mrgenixus will you please add screenshots to the description? Thanks

File: spec/features/mentorships_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+feature "I want to see a list of past mentorships", focus: true do
```
Comment (rebeccafae) [inline spec/features/mentorships_index_spec.rb]: focus

File: spec/factories/mentorships.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 FactoryGirl.define do
   factory :mentorship do
+    mentor
```
Comment (rebeccafae) [inline spec/factories/mentorships.rb]: does this break existing tests?

File: app/views/mentorships/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            <% else %>
+              <i>Pending</i>
+              <%= link_to "Cancel", mentorship_path(mentorship), method: :delete %></br>
+            <% end %>
+          </td>
+          <td class="mentorship--mentee">
+            <% if mentorship.mentee == current_user %>
+              <%= current_user.full_name %>
+            <% else %>
+              <%= link_to mentorship.mentee.full_name, user_path(mentorship.mentee) %>
```
Comment (rebeccafae) [inline app/views/mentorships/index.html.erb:25]: Did we confirm if the current_user will have access to this user's page?

**Review by rebeccafae — CHANGES_REQUESTED** [review]

File: app/views/mentorships/index.html.erb (hunk 3 above)
Comment (mrgenixus) [reply app/views/mentorships/index.html.erb:25]: Oh; I didn't clarify that with Kate, but I should add that; thanks. Good Catch

File: spec/factories/mentorships.rb (hunk 2 above)
Comment (mrgenixus) [reply spec/factories/mentorships.rb]: I don't believe so, but I'm noticing the 'focus' and think that I have maybe not checked

File: spec/features/mentorships_index_spec.rb (hunk 1 above)
Comment (mrgenixus) [reply spec/features/mentorships_index_spec.rb]: indeed

File: app/views/mentorships/index.html.erb (hunk 3 above)
Comment (rebeccafae) [reply app/views/mentorships/index.html.erb:25]: This has already been clarified with Kate in the past, we want them to be able to see their mentors/mentees. I just don't remember how we wrote the code.

Comment (mrgenixus) [reply app/views/mentorships/index.html.erb:25]: I could use 5 minutes of help with this

**Comment by mrgenixus** [comment]
@rebeccafae please re-review

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,25 +1,21 @@
 class UsersController < ApplicationController
   before_filter :auth_user
+  before_filter :set_industries_and_interests, only: [:edit, :update]
```
Comment (rebeccafae) [inline app/controllers/users_controller.rb:3]: 👍

**Review by rebeccafae — APPROVED** [review]

File: app/controllers/mentorships_controller.rb [hunk 5]
```diff
@@ -7,13 +7,13 @@ def auth_user
...
     @user = current_user
-    @mentorship = Mentorship.new()
+    @mentorship = Mentorship.new
     @industries = ['Technology', 'Business', 'Startup']
   end
 
   def create
     @user = current_user
-    @mentorship = Mentorship.new(mentee_id: current_user.id, question: params[:mentorship][:question])
+    @mentorship = Mentorship.new mentee_id: current_user.id, question: params[:mentorship][:question]
```
Comment (rebeccafae) [inline app/controllers/mentorships_controller.rb]: let's keep parenthesis here, to stay consistent with [style guide](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens)

**Pull request merged**